src/psyclone/psyir/nodes/structure_reference.py

```diff
@@ -199,6 +220,8 @@ def _create(cls, symbol, symbol_type, members, parent=None):
             child_member = subref
         # Finally, add this chain to the top-level reference
         ref.addchild(child_member)
+        # pylint: disable=attribute-defined-outside-init
```

Comment:
It may still complain about accessing a private member, this is OK, I prefer to keep this private for now, you can disable this one.

Reply:
I hope what I did was what you wanted.

Reply:
Almost, StructureReference should still have a positional "symbol" parameter which is just passed as a positional parameter to the super(). kwargs doesn't keep it (I think).

(it can also be done with an *args parameter but since it is just one maybe "symbol" end up with better documentation, but I don't mind which way it is implemented)

Also a constructor test in src/psyclone/tests/psyir/nodes/structure_reference_test.py would be good, even if the code is already covered indirectly.

Reply:
OK, I added 'symbol' as positional parameter back in.

Reply:
I also added a small test for the constructor.